jib-core/src/main/java/com/google/cloud/tools/jib/frontend/JavaLayerConfigurations.java

```diff
@@ -126,6 +126,44 @@ public Builder addDirectoryContents(
       return this;
     }
 
+    /**
+     * Adds directory contents to a layer selectively (via {@code pathFilter}) and recursively.
+     * {@code sourceRoot} must be a directory. Empty directories will always be added regardless of
+     * {@code pathFilter}, except that {@code sourceRoot} is never added. Permissions are specified
+     * via {@code permissionsMap}, which maps from extraction path on the container to file
+     * permissions.
+     *
+     * @param sourceRoot root directory whose contents will be added
+     * @param basePathInContainer directory in the layer into which the source contents are added
+     * @param permissionsMap the map from absolute path on container to file permission
+     * @return this
+     * @throws IOException error while listing directories
+     * @throws NotDirectoryException if {@code sourceRoot} is not a directory
+     */
+    // TODO: Use in plugins
+    public Builder addExtraDirectoryContents(
```

Comment:
Same logic here as in the comment for the permissions-vs.-no-permissions methods. I think this can just be `addDirectoryContents` as well (we'll only use it for extra directories), and then the other `addDirectoryContents` would just call this one with no permissions to set, so that the code duplication can be reduced.

Reply:
I think for this method, we should also be setting the same permissions for all subfiles of a directory if a permission was specified for that directory?

Reply:
>I think this can just be addDirectoryContents as well ...

I felt the need to specify "Extra" in the name since I got rid of the `LayerType` parameter. But yeah I think we can just make it an overload that can take an empty permissions map.